OpenRA.Mods.Common/Scripting/Global/ActorGlobal.cs

```diff
@@ -71,7 +71,20 @@ public int BuildTime(string type)
 			if (!Context.World.Map.Rules.Actors.TryGetValue(type, out ai))
 				throw new LuaException("Unknown actor type '{0}'".F(type));
 
-			return ai.GetBuildTime();
+			var bi = ai.TraitInfoOrDefault<BuildableInfo>();
+
+			if (bi == null)
+				return 0;
+
+			var time = bi.BuildDuration;
+			if (time == -1)
+			{
+				var valued = ai.TraitInfoOrDefault<ValuedInfo>();
+				time = valued != null ? valued.Cost : 0;
+			}
+
+			time = time * bi.BuildDurationModifier / 100;
```

Comment:
TBH, `ProductionQueue.BuildDurationModifier` is ignored here even in bleed.

@pchote: If you try to force me to get it removed again, then I'm gonna close this PR. I'm not gonna work against myself. The per-queue modifier existed in all the WW games while the per-unit one arrived in YR.